DataAnnotationsValidator/DataAnnotationsValidator.Tests/DataAnnotationsValidatorTests.cs

```diff
@@ -157,6 +157,29 @@ public void TryValidateObject_modifies_membernames_for_nested_properties()
 			Assert.AreEqual(1, validationResults.Count);
 			Assert.AreEqual("Child PropertyA is required", validationResults[0].ErrorMessage);
 			Assert.AreEqual("Child.PropertyA", validationResults[0].MemberNames.First());
-		}
+		}
+
+        [Test]
+        public void TryValidateObject_object_with_dictionary_does_not_fail()
+        {
+            var classWithDictionary = new ClassWithDictionary
+            {
+                Objects =
+                    new List<Dictionary<string, Child>>{
+                    new Dictionary<string, Child>
+                {
+                    { "key", new Child
+                    {
+                        PropertyA = 1,
+                    PropertyB = 2
+                    } } } }

```

Comment:
can you clean up the construction of this class a bit? all the nesting leaves this a bit gnarly.